src/Views/AddAccountView.vala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2013-2018 elementary, Inc. (https://elementary.io)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ * Authored by: Corentin Noël <corentin@elementary.io>
+ */
+
+public class OnlineAccounts.AddAccountView : Gtk.Grid {
+    construct {
+        var primary_label = new Gtk.Label (_("Connect Your Online Accounts"));
+        primary_label.wrap = true;
+        primary_label.max_width_chars = 60;
+        primary_label.xalign = 0;
+        primary_label.get_style_context ().add_class (Granite.STYLE_CLASS_H2_LABEL);
+
+        var secondary_label = new Gtk.Label (_("Sign in to connect with apps like Mail, Contacts, and Calendar."));
+        secondary_label.wrap = true;
+        secondary_label.max_width_chars = 60;
+        secondary_label.xalign = 0;
+
+        var listbox = new Gtk.ListBox ();
+        listbox.activate_on_single_click = true;
+        listbox.vexpand = true;
+
+        var manager = new Ag.Manager ();
+        foreach (var provider in manager.list_providers ()) {
+            if (provider == null || provider.get_plugin_name () == null) {
+                continue;
+            }
+            listbox.add (new AccountRow (provider));
+        }
+
+        var scrolled_window = new Gtk.ScrolledWindow (null, null);
+        scrolled_window.hscrollbar_policy = Gtk.PolicyType.NEVER;
+        scrolled_window.add (listbox);
+
+        var frame = new Gtk.Frame (null);
+        frame.margin_top = 24;
+        frame.add (scrolled_window);
+
+        halign = Gtk.Align.CENTER;
+        margin = 12;
+        orientation = Gtk.Orientation.VERTICAL;
```

Comment:
put the properties of the said object at the beginning of the construct {} (personal convention)